lib/main.coffee

```diff
@@ -49,14 +54,17 @@ module.exports =
         filePath = textEditor.getPath()
         command = @executablePath
         ruleset = @rulesets
-        if @projectRules
-          rulesetPath = helpers.find(filePath, 'ruleset.xml')
+        if /^[a-z0-9]+\.xml$/gi.test(@rulesets)
+          rulesetPath = helpers.find(filePath, @rulesets)
           ruleset = rulesetPath if rulesetPath?
         parameters = []
         parameters.push(filePath)
         parameters.push('text')
         parameters.push(ruleset)
-        return helpers.exec(command, parameters).then (output) ->
+        options = {}
+        projectDir = atom.project.relativizePath(filePath)[0]
+        options.cwd = projectDir or textEditor.getDirectoryPath()
```

Comment:
I should've checked the docs for that one, to be honest I found it while reverse engineering how Atom change the titlebar if you open two files with the same filename but in different projects:
![skjermbilde 2016-06-01 kl 19 29 01](https://cloud.githubusercontent.com/assets/81981/15719239/547f6334-282f-11e6-917c-3c5becf0e49a.png)

I found it here: https://github.com/atom/atom/blob/1c843e70a21aa1e496c0d7cf7d6542148019445a/src/text-editor.coffee#L700

Maybe it's best to assume that if it's not in the docs it's private and can change without notice?